SmartDeviceLink/private/SDLProtocolReceivedMessageProcessor.m

```diff
@@ -0,0 +1,317 @@
+            
+            // Check for errors.
+            // For these two frame types, the frame info should be 0x00
+            if ((frameType == SDLFrameTypeFirst) || (frameType == SDLFrameTypeSingle)){
+                if (controlFrameInfo != 0x00) {
+                    prevState=state;
+                    state = ERROR_STATE;
+                }
+            }
+            break;
+            
+        case SESSION_ID_STATE:
+            // Add the byte to the headerBuffer
+            [self.headerBuffer appendBytes:&currentByte length:sizeof(currentByte)];
+            
+            // Set the next state
+            state = DATA_SIZE_1_STATE;
+            break;
+        
+        // 32 bits for data size
+        case DATA_SIZE_1_STATE:
+            dataLength = 0;
+            dataLength += (currentByte & 0xFF ) << 24;
+            state = DATA_SIZE_2_STATE;
+            // Add the byte to the headerBuffer
+            [self.headerBuffer appendBytes:&currentByte length:sizeof(currentByte)];
+            break;
+            
+        case DATA_SIZE_2_STATE:
+            dataLength += (currentByte & 0xFF ) << 16;
+            state = DATA_SIZE_3_STATE;
+            // Add the byte to the headerBuffer
+            [self.headerBuffer appendBytes:&currentByte length:sizeof(currentByte)];
+            break;
+            
+        case DATA_SIZE_3_STATE:
+            dataLength += (currentByte & 0xFF ) << 8;
+            state = DATA_SIZE_4_STATE;
+            // Add the byte to the headerBuffer
+            [self.headerBuffer appendBytes:&currentByte length:sizeof(currentByte)];
+            break;
+            
+        case DATA_SIZE_4_STATE:
+            dataLength += (currentByte & 0xFF ) << 0;
+            
+            // Set the next state
+            state = MESSAGE_1_STATE;
+            
+            // Add the byte to the headerBuffer
+            [self.headerBuffer appendBytes:&currentByte length:sizeof(currentByte)];
+            
+            // Set the counter for the data pump.
+            dataBytesRemaining = dataLength;
+            
+            // Version 1 does not have a message ID so we skip to the data pump, or the end.
+            if( version == 1){
+                if (dataLength == 0) {
+                    [self ResetState];
+                } else {
+                    // skip ahead to the data pump state
+                    state = DATA_PUMP_STATE;
+                }
+            }
+            
+            Byte headerSize = 0;
+            // Figure out headerSize
+            if( version == 1){
+                headerSize = 8;
+            }else{
+                headerSize = 12;
+            }
+            
+            int maxMtuSize = 0;
+            // Figure out max MTU size
+            if (version <= 2){
+                maxMtuSize = 1500;
+            } else{
+                maxMtuSize = 131084;
+            }
+            
+            // Check data length (does it conform to spec?)
```

Comment:
This comment is obvious, but the code is not. Clarify the comment by saying _why_ you are checking `dataLength >= (maxMtuSize - headerSize)`